### DISCUSSION TITLE
How to use multiple form in the `ModelResource` class ?

### DISCUSSION THREAD

**Discussion by detygon** [filamentphp/filament#993] [Help]
It's currently possible to use multiple forms on the same page when using the form builder but there is not a documented way to do the same using the `ModelResource` class. My use case is that I would like to have the functionality of the relationship manager on the create page.

**Comment by danharrin** [comment]
Hi, if you'd like to create HasMany records from the create page I'd recommend a `HasManyRepeater`.

**Reply by detygon** [reply]
Thanks, is there a way to do the same for a belongs-to relationship? Like a way to dynamically create an item and link it to the record being created.

**Comment by kjlloydie** [comment]
I'd also like to know if there's support for using multiple forms with the Resource classes, so that I can include forms for creating records in BelongsTo relationships (in lieu of in-line relation creation, like [Nova supports](https://nova.laravel.com/docs/3.0/resources/relationships.html#creating-inline-relations)).

**Reply by ryangjchandler** [reply]
Inline record creation for BelongsTo is coming soon - #2057.

I wonder if we could perhaps have a `HasOneRelationManager` class that would allow you to edit `HasOne` relations inline.

**Reply by kjlloydie** [reply]
Nice idea, thanks for the inspiration! I threw together a [BelongsToRelationManager.php](https://gist.github.com/kjlloydie/badc4d87cf0e4477a90af445c96c8a72) as a temporary measure for my own purposes, until in-line support is ready. I've only just started using Filament though, so I can speak to the robustness of this solution.

**Reply by ryangjchandler** [reply]
> Nice idea, thanks for the inspiration! I threw together a [BelongsToRelationManager.php](https://gist.github.com/kjlloydie/badc4d87cf0e4477a90af445c96c8a72) as a temporary measure for my own purposes, until in-line support is ready. I've only just started using Filament though, so I can speak to the robustness of this solution.

I'll put it I my list of things to experiment with.
